trunk/apps/dir2hdf5.py

```diff
@@ -0,0 +1,118 @@
+
+import argparse
+import sys
+import os
+import os.path as osp
+
+import h5py
+import numpy as np
+
+from nn.profiler import setupLogging
+from dataset.reader import readImage
+
+STORAGE_TYPE = np.uint8
+
+def dir2hdf5(directory, outputfile, convert, log):
+    ''' Convert a directory to an hdf5 file with the same content '''
+
+    log.info('Processing {0} into {1}'.format(directory, outputfile))
+
+    with h5py.File(outputfile, 'a') as h5file:
+
+        fidx = 0
+        for root, dirs, files in os.walk(directory):
+
+            # Create groups for directories
+            for d in (osp.join(root, d) for d in dirs):
+                dset = h5file.create_group(d)
+
+            for f in (osp.join(root, f) for f in files):
+
+                # replace old contents with new
+                if f in h5file:
+                    del h5file[f]
+                if convert:
+                    blob, typeinfo = readImage(f, None, raw=True)
+                else:
+                    blob = np.fromfile(f, dtype=STORAGE_TYPE)
```

Comment:
Instead of having this as a special case, we should just add a readNP or something to dataset.reader so this type is officially supported. You will them be able to remove this logic and call openImage() directly to get the raw pixel data